docs/typescript/patching.md

```diff
@@ -14,7 +14,55 @@ Before you explore our patching/versioning API, check if your needs can be addre
 1. Remove `Workflow1` code when none of them are running anymore
 
 This means Workflows running `Workflow1` code will never migrate to `Workflow2` code before they complete.
-If you would like to update `Workflow1` code while it is running, you will need to "patch in" code. Read on.
+If you would like to update `Workflow1` code while it is running, you may need to "patch in" code. Read on.
+
+## Do I need to Patch?
+
+You may need to patch if:
+
+- You want to change the remaining logic of a Workflow while it is still running
+- If your new logic can result in a different execution path
+
+This code does not need patching (i.e. since execution path does not change, you can update your Workflow and restart Workers straight away):
+
+```ts
+// from v1
+export async function myWorkflow({ force }): Promise<void> {
+  await sleep('1 days');
+  console.log('force', force);
+}
+
+// to v2
+export async function myWorkflow(payload): Promise<void> {
+  let force = payload.force;
+
+  setListener(updatePayloadSignal, (newPayload) => {
+    force = newPayload.force;
+  });
+
+  await sleep('1 days');
+  console.log('force', force);
+}
+```
+
+This is an example of code that will need patching:
```

Comment:
This is more an in your face example of how npm packages might break determinism.

Reply:
```suggestion
This is an example of how npm packages can break determinism:
```